lib/gltfStatistics.js

```diff
@@ -0,0 +1,202 @@
+    var numberOfDrawCalls = 0;
+    var numberOfRenderedPrimitives = 0;
+
+    var nodes = gltf.nodes;
+    var meshes = gltf.meshes;
+    var accessors = gltf.accessors;
+
+    var nodeStack = [];
+
+    var n = nodes[nodeId];
+    nodeStack.push(n);
+
+    while (nodeStack.length > 0) {
+        n = nodeStack.pop();
+
+        var nodeNeshes = defaultValue(n.meshes, defined(n.instanceSkin) ? n.instanceSkin.meshes : undefined);
+        if (defined(nodeNeshes)) {
+            var meshesLength = nodeNeshes.length;
+            for (var j = 0; j < meshesLength; ++j) {
+                var primitives = meshes[nodeNeshes[j]].primitives;
+                var primitivesLength = primitives.length;
+                for (var m = 0; m < primitivesLength; ++m) {
+                    var primitive = primitives[m];
+                    var indices = primitive.indices;
+                    var indicesCount = accessors[indices].count;
+
+                    switch(primitive.primitive) {
+                        case 0: // POINTS
+                            numberOfRenderedPrimitives += indicesCount;
+                            break;
+                        case 1: // LINES
+                            numberOfRenderedPrimitives += (indicesCount / 2);
+                            break;
+                        case 2: // LINE_LOOP
+                            numberOfRenderedPrimitives += indicesCount;
+                            break;
+                        case 3: // LINE_STRIP
+                            numberOfRenderedPrimitives += Math.max(indicesCount - 1, 0);
+                            break;
+                        case 4: // TRIANGLES
+                            numberOfRenderedPrimitives += (indicesCount / 3);
+                            break;
+                        case 5: // TRIANGLE_STRIP
+                        case 6: // TRIANGLE_FAN
+                            numberOfRenderedPrimitives += Math.max(indicesCount - 2, 0);
+                            break;
+                    }
+                }
+                numberOfDrawCalls += primitivesLength;
+            }
+        }
+
+        var children = n.children;
+        var childrenLength = children.length;
+        for (var k = 0; k < childrenLength; ++k) {
+            var child = nodes[children[k]];
+            nodeStack.push(child);
+        }
+    }
+
+    return {
+        numberOfDrawCalls : numberOfDrawCalls,
+        numberOfRenderedPrimitives : numberOfRenderedPrimitives
+    };
+}
+
+function getAllDrawCallStats(gltf) {
+    var numberOfDrawCalls = 0;
+    var numberOfRenderedPrimitives = 0;
+
+    var rootNodes = gltf.scenes[gltf.scene].nodes;
+    var rootNodesLength = rootNodes.length;
+
+    for (var i = 0; i < rootNodesLength; ++i) {
+        var stats = getDrawCallStatistics(gltf, rootNodes[i]);
+        numberOfDrawCalls += stats.numberOfDrawCalls;
+        numberOfRenderedPrimitives += stats.numberOfRenderedPrimitives;
+    }
```

Comment:
I would be fine with removing `getDrawCallStatistics` and just placing that logic in here. Then the code can be heavily condensed heavily by using `PrimitiveHelpers.getAllPrimitives`. Then `getAllDrawCallStats` can take the name `getDrawCallStatistics`.